sympy/sets/tests/test_conditionset.py

```diff
@@ -85,30 +99,35 @@ def test_free_symbols():
         ).free_symbols == {z}
     assert ConditionSet(x, Eq(x, 0), FiniteSet(x, z)
         ).free_symbols == {x, z}
-    assert ConditionSet(x, Eq(x, 0), ImageSet(Lambda(y, y**2), S.Integers)
-        ).free_symbols == set()
+    assert ConditionSet(x, Eq(x, 0), ImageSet(Lambda(y, y**2),
+        S.Integers)).free_symbols == set()
 
 
 def test_bound_symbols():
     assert ConditionSet(x, Eq(y, 0), FiniteSet(z)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, Eq(x, 0), FiniteSet(x, y)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, x < 10, ImageSet(Lambda(y, y**2), S.Integers)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
     assert ConditionSet(x, x < 10, ConditionSet(y, y > 1, S.Integers)
-        ).bound_symbols == {x}
+        ).bound_symbols == [x]
```

Comment:
Is there a reason for bound symbols to be a list rather than a set in general?

Reply:
Yes, so a canonical expression can be obtained from `as_dummy`, e.g. 
```
C = ConditionSet
C((x,y),x<y,Reals).as_dummy() == C((y,x),y<x,Reals).as_dummy()
```
The symbols are numbered by their physical location rather than their names.